### PR TITLE
xapian: 1.4.19 -> 1.4.20

### PR DESCRIPTION
**Pull request by chvp**
###### Description of changes

https://trac.xapian.org/wiki/ReleaseOverview/1.4.20

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by peterhoeg** [comment]
@GrahamcOfBorg build mu

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 181467` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>11 packages marked as broken and skipped:</summary>
  <ul>
    <li>dovecot_fts_xapian</li>
    <li>gmailieer</li>
    <li>libsForQt5.appstream-qt</li>
    <li>libsForQt512.appstream-qt</li>
    <li>libsForQt515.appstream-qt</li>
    <li>python310Packages.notmuch2</li>
    <li>python310Packages.recoll</li>
    <li>python39Packages.notmuch2</li>
    <li>python39Packages.recoll</li>
    <li>recoll</li>
    <li>setzer</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.xapian</li>
    <li>python39Packages.xapian</li>
  </ul>
</details>
<details>
  <summary>20 packages built:</summary>
  <ul>
    <li>aerc</li>
    <li>afew</li>
    <li>appstream</li>
    <li>lbdb</li>
    <li>lieer</li>
    <li>mu</li>
    <li>muchsync</li>
    <li>mujmap</li>
    <li>neomutt</li>
    <li>notmuch</li>
    <li>notmuch-addrlookup</li>
    <li>perl532Packages.SearchXapian</li>
    <li>perl534Packages.SearchXapian</li>
    <li>public-inbox</li>
    <li>python310Packages.notifymuch</li>
    <li>python310Packages.notmuch</li>
    <li>python39Packages.notifymuch</li>
    <li>python39Packages.notmuch</li>
    <li>xapian</li>
    <li>xapian-omega</li>
  </ul>
</details>

**Comment by chvp** [comment]
@siraben interesting that the python packages fail for you, do you have any more detail on the failure? (They build on my x86_64-linux machine.) Do they build before this patch?

**Comment by chvp** [comment]
@GrahamcOfBorg build python310Packages.xapian

**Comment by chvp** [comment]
Seems like it was already failing on aarch64-darwin before this PR: https://hydra.nixos.org/build/183130871

File: pkgs/development/libraries/xapian/default.nix [hunk 1]
```diff
@@ -38,5 +38,8 @@ let
     };
   };
 in {
-  xapian_1_4 = generic "1.4.19" "sha256-H8pI/KbMNSbMS6k90ZT+nBMmhXt47c+zfmjQhtcUqcM=";
+  # Don't forget to change the hashes in xapian-omega and
+  # python3Packages.xapian. They inherit the version from this package, and
+  # should always be built with the equivalent xapian version.
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/xapian/default.nix:43]: 👍🏼

File: pkgs/development/libraries/xapian/tools/omega/default.nix [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+{ lib, stdenv, fetchurl, pkg-config, xapian, perl, pcre2, zlib, libmagic }:
 
 stdenv.mkDerivation rec {
   pname = "xapian-omega";
   inherit (xapian) version;
 
   src = fetchurl {
     url = "https://oligarchy.co.uk/xapian/${version}/xapian-omega-${version}.tar.xz";
-    sha256 = "0zji8ckp4h5xdy2wbir3lvk680w1g1l4h5swmaxsx7ah12lkrjcr";
+    sha256 = "Cf19bGCzlP0A2EcAZJlpofzDqiuIuyxsoiD9+o0liB8=";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/xapian/tools/omega/default.nix:9]:
```suggestion
    sha256 = "sha256-Cf19bGCzlP0A2EcAZJlpofzDqiuIuyxsoiD9+o0liB8=";
```
or
```suggestion
    hash = "sha256-Cf19bGCzlP0A2EcAZJlpofzDqiuIuyxsoiD9+o0liB8=";
```

File: pkgs/development/python-modules/xapian/default.nix [hunk 3]
```diff
@@ -13,7 +13,7 @@ buildPythonPackage rec {
 
   src = fetchurl {
     url = "https://oligarchy.co.uk/xapian/${version}/xapian-bindings-${version}.tar.xz";
-    sha256 = "0j9awiiw9zf97r60m848absq43k37gghpyw7acxqjazfzd71fxvm";
+    sha256 = "eGzCjQVmCyJ5VEE68OL2bk6tKgbT322rrqSERUtgHvU=";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/xapian/default.nix:16]:
```suggestion
    sha256 = "sha256-eGzCjQVmCyJ5VEE68OL2bk6tKgbT322rrqSERUtgHvU=";
```
same here

**Comment by chvp** [comment]
Yeah, I was able to find the log on hydra and ofborg. It's not a new failure and seems pretty darwin-specific, I don't really have any idea on how to fix it (did quite a bit of searching, but couldn't really find anyone having similar problems).

**Comment by peterhoeg** [comment]
> 2 packages failed to build:
>
> * python310Packages.xapian
> * python39Packages.xapian

The python bindings failing is a problem.

Has anybody tried on x64?

**Comment by chvp** [comment]
Seems like it builds correctly on x86_64: https://hydra.nixos.org/job/nixpkgs/trunk/python310Packages.xapian.x86_64-darwin/all

**Comment by peterhoeg** [comment]
I'm sorry about bringing this back to life. It looks like I initially replied back in July via email to github but it's only *now* adding it as a comment and thus necro-bumping the issue.

**Comment by peterhoeg** [comment]
It's happened to 6 issues so far over the last 30 minutes. Good times.

**Pull request merged**